File: PentagonNumbers/PN.py
#!/usr/bin/env python3

"""
Project Euler # 44

"""

from math import sqrt

def is_pentagonal(n):
    x = (sqrt(24*n + 1) + 1)/6
    if (x).is_integer():
        return True
    else:
        return False

def pentagonal(n):
    return int(n*(3*n -1)/2)

n = 2
switch = True

while switch:
    n += 1
    p_k = pentagonal(n)

    for i in reversed(range(1, n-1)):
        p_j = pentagonal(i)
        if (is_pentagonal(p_k-p_j) and is_pentagonal(p_k+p_j)):
            print((n,p_k), (i,p_j), (p_k - p_j))
            switch = False
            break

File: PentagonNumbers/PN.java
/*
Project Euler # 44

*/

public class PN {
    public static void main( String[] args ) {
        int n = 2;
        boolean flag = true;

        while ( flag ) {
            n++;
            int pk = pentagonal(n);

            for ( int i=n-1; i>0; i-- ) {
                int pj = pentagonal(i);
                if ( is_pentagonal(pk-pj) && is_pentagonal(pk+pj) ) {
                    System.out.println(pk-pj);
                    flag = false;
                    break;
                }
            }
        }
    }

    public static int pentagonal( int n ) {
        return (int) n * (3*n - 1)/2;
    }

    public static boolean is_pentagonal( int n ) {
        double x = (Math.sqrt(24*n + 1) + 1)/6;

        if ( x == (int)x )
            return true;
        else
            return false;
    }
}
